pkg/providers/snow/creds.go

```diff
@@ -4,25 +4,21 @@ import (
 	"fmt"
 
 	"github.com/aws/eks-anywhere/pkg/aws"
+	"github.com/aws/eks-anywhere/pkg/cluster"
 )
 
-type bootstrapCreds struct {
-	snowCredsB64 string
-	snowCertsB64 string
-}
-
-func (p *SnowProvider) setupBootstrapCreds() error {
+func (p *SnowProvider) setupBootstrapCreds(clusterSpec *cluster.Spec) error {
 	creds, err := aws.EncodeFileFromEnv(eksaSnowCredentialsFileKey)
 	if err != nil {
 		return fmt.Errorf("failed to set up snow credentials: %v", err)
 	}
-	p.bootstrapCreds.snowCredsB64 = creds
 
 	certs, err := aws.EncodeFileFromEnv(eksaSnowCABundlesFileKey)
 	if err != nil {
 		return fmt.Errorf("failed to set up snow certificates: %v", err)
 	}
-	p.bootstrapCreds.snowCertsB64 = certs
+
+	clusterSpec.SnowCredentialsSecret = EksaCredentialsSecret(clusterSpec, creds, certs)
```

Comment:
And what step of the CLI flow does this happen?

Reply:
Ahhh I see it now
I would vote to make this line part of `configManager.SetDefaultsAndValidate` (the defaults part)